internal/updater/github.go

```diff
@@ -0,0 +1,97 @@
+package updater
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+	"strings"
+
+	"github.com/dominikschulz/github-releases/ghrel"
+	"github.com/gopasspw/gopass/pkg/debug"
+	"github.com/pkg/errors"
+)
+
+const (
+	gitHubOrg  = "gopasspw"
+	gitHubRepo = "gopass"
+)
+
+func filterMajor(rs []ghrel.Release, major uint64) []ghrel.Release {
+	out := make([]ghrel.Release, 0, len(rs)-1)
+	for _, r := range rs {
+		v := r.Version()
+		if v.Major != major {
+			continue
+		}
+		out = append(out, r)
+	}
+	return out
+}
+
+func findAssets(r ghrel.Release) (string, string) {
+	var urlTar string
+
+	debug.Log("Assets: %+v", r.Assets)
+	for _, asset := range r.Assets {
+		name := strings.TrimSuffix(strings.TrimPrefix(asset.Name, "gopass-"), ".tar.gz")
+		p := strings.Split(name, "-")
+		if len(p) < 3 {
+			continue
+		}
+		if p[len(p)-2] != runtime.GOOS {
+			continue
+		}
+		if p[len(p)-1] != runtime.GOARCH {
+			continue
+		}
+		if asset.URL == "" {
+			continue
+		}
+		urlTar = asset.URL
+		break
+	}
+
+	for _, asset := range r.Assets {
+		if !strings.HasSuffix(asset.Name, "SHA256SUMS") {
+			continue
+		}
+		return urlTar, asset.URL
+	}
+
+	return "", ""
+}
+
+func simpleUpdate(ctx context.Context, r ghrel.Release) error {
+	uTar, uSha := findAssets(r)
+	if uTar == "" {
+		return errors.New("no supported binary found")
+	}
+	if uSha == "" {
+		return errors.New("no checksum published")
+	}
+	// TODO download and check sig
+	if err := updateTo(ctx, r.Version().String(), uTar); err != nil {
+		return errors.Wrapf(err, "Failed to update gopass: %s", err)
+	}
+	return nil
+}
+
+// FetchReleases fetches and returns all releases of gopass from GitHub
+func FetchReleases(pre bool) ([]ghrel.Release, error) {
+	if pre {
+		return ghrel.FetchAllReleases(gitHubOrg, gitHubRepo)
+	}
+	return ghrel.FetchAllStableReleases(gitHubOrg, gitHubRepo)
+}
+
+// LatestRelease fetches and returns the latest gopass release from GitHub
+func LatestRelease(pre bool) (ghrel.Release, error) {
```

Comment:
This is available in https://github.com/dominikschulz/github-releases/blob/master/ghrel/releases.go#L111, no?

Reply:
No, I'm ditching that for now.